zerver/lib/message.py

```diff
@@ -1756,9 +1758,20 @@ def check_user_group_mention_allowed(sender: UserProfile, user_group_ids: List[i
     user_groups = UserGroup.objects.filter(id__in=user_group_ids).select_related(
         "can_mention_group"
     )
+    sender_is_system_bot = is_cross_realm_bot_email(sender.delivery_email)
 
     for group in user_groups:
         can_mention_group = group.can_mention_group
+
+        if sender_is_system_bot:
+            if can_mention_group.name == SystemGroups.EVERYONE:
+                continue
+            raise JsonableError(
+                _(
+                    "You are not allowed to mention user group '{user_group_name}'. You must be a member of '{can_mention_group_name}' to mention this group."
+                ).format(user_group_name=group.name, can_mention_group_name=can_mention_group.name)
```

Comment:
Have added this so that we can avoid calling `is_user_in_group` for system bots as we already know that they are not part of the system groups. We might also update the error message here to not have "You must be a member of .." line since we would anyways not allow system bots to join groups in other realm.

Reply:
This is probably fine as a stopgap but the long-term fix is likely to copy system bots into each realm.